Add more docs (#539)
DIRACGrid/diracx: docs/dev/explanations/components/index.md

# Packaging and versioning

> DiracX is written in Python package and composed of several interconnected submodules.

Reviewer: ```suggestion
DiracX is written in Python. As a package, it is composed of several interconnected submodules.
```

It's designed to provide robust and versatile functionalities, primarily through these key components:

1. **User-Facing Components**:

    - **`diracx`**: This is the primary interface for users, integrating both the Command Line Interface (CLI) and Python API.
    - **`diracx-routers`**: Serves as the server component, offering HTTPS endpoints.

>     - **`diracx-tasks`**: Handles operations executed by DiracX servers, either periodically or upon request.

Reviewer: This is not "User-facing".

1. **Base Image**:

    - All container images start from `diracx/base`.

2. **Specialized Base Images**:

    - `diracx/services-base`
    - `diracx/tasks-base`
    - `diracx/client-base`

3. **Image Versioning and Building**:

    - Images are built periodically (e.g., every Monday) and tagged as `YYYY.MM.DD.P`.

>     - A DiracX release triggers the creation of new `DiracXService`, `diracx/tasks`, and `diracx/client` images, based on specific `diracx/base` tags.

Reviewer: ```suggestion
    - A DiracX release triggers the creation of new `diracx/service`, `diracx/tasks`, and `diracx/client` images, based on specific `diracx/base` tags.
```
?

TODO: document the entry point

- `diracx-routers`:

>     - `diracx.diracx_min_client_version` entry-point defines the diracx minimum client version required by the server to prevent issues. This also searches for extension names instead of `diracx`. The minimum version number has to be updated in `diracx-routers/src/__init.py__`

Reviewer: ```suggestion
    - `diracx.diracx_min_client_version` entry-point defines the DiracX minimum client version required by the server to prevent issues. This also searches for extension names instead of `diracx`. The minimum version number has to be updated in `diracx-routers/src/__init.py__`
```

and specify what is an acceptable extension name.
